### PR TITLE
Remove obsolete YAML test cases

### PR DESCRIPTION
**Pull request by mosche**
Follow up on https://github.com/elastic/elasticsearch/pull/108055

### CODE REVIEW

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.put_settings/all_path_options.yml [hunk 1]
```diff
@@ -78,24 +78,6 @@ setup:
   - match: {test_index2.settings.index.refresh_interval:     10s}
   - is_false: foo.settings.index.refresh_interval
 
----
-"put settings in list of indices":
```
Comment (mosche) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.put_settings/all_path_options.yml:82]: Back from 2014, feature was never implemented

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/rollup/get_jobs.yml [hunk 2]
```diff
@@ -98,129 +98,3 @@ setup:
 
   - match:
       jobs: []
-
----
```
Comment (mosche) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/rollup/get_jobs.yml:102]: Rollup will be removed anyways

Comment (thecoop) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/rollup/get_jobs.yml:102]: Rollup hasn't been removed yet though...

Comment (mosche) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/rollup/get_jobs.yml:102]: yes, but with that in mind i don't think anyone will fix that muted test, right?

Comment (mosche) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/rollup/get_jobs.yml]: Moved this one to separate pr

**Review by ldematte — APPROVED** [review]
👍

**Pull request merged**